sherpa/cpp_api/grpc/online-grpc-client.cc

```diff
@@ -0,0 +1,150 @@
+// sherpa/cpp_api/grpc/online-grpc-client.cc
+//
+// Copyright (c) 2023 y00281951
+
+#include <chrono>  // NOLINT
+#include <random>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <cstdlib>
+
+#include "grpc/grpc.h"
+#include "grpcpp/channel.h"
+#include "grpcpp/client_context.h"
+#include "grpcpp/create_channel.h"
+
+#include "sherpa/csrc/wav.h"
+#include "sherpa/csrc/log.h"
+#include "sherpa/csrc/file-utils.h"
+#include "sherpa/cpp_api/parse-options.h"
+#include "sherpa/cpp_api/grpc/online-grpc-client-impl.h"
+
+static constexpr const char *kUsageMessage = R"(
+Automatic speech recognition with sherpa using grpc.
+
+Usage:
+
+sherpa-online-grpc-client --help
+
+sherpa-online-grpc-client \
+  --server-ip=127.0.0.1 \
+  --server-port=6006 \
+  --wav-path=/path/to/foo.wav
+
+or
+
+sherpa-online-grpc-client \
+  --server-ip=127.0.0.1 \
+  --server-port=6006 \
+  --wav-scp=/path/to/wav.scp
+)";
+
+void SplitStringToVector(const std::string &full, const char *delim,
+                         bool omit_empty_strings,
+                         std::vector<std::string> *out) {
+  size_t start = 0, found = 0, end = full.size();
+  out->clear();
+  while (found != std::string::npos) {
+    found = full.find_first_of(delim, start);
+    // start != end condition is for when the delimiter is at the end
+    if (!omit_empty_strings || (found != start && start != end))
+      out->push_back(full.substr(start, found - start));
+    start = found + 1;
+  }
+}
+
+int32_t main(int32_t argc, char* argv[]) {
+  std::string server_ip = "127.0.0.1";
+  int32_t server_port = 6006;
+  std::string wav_path = "";
+  std::string wav_scp = "";
+  sherpa::ParseOptions po(kUsageMessage);
+
+  po.Register("server-ip", &server_ip, "IP address of the grpc server");
+  po.Register("server-port", &server_port, "Port of the grpc server");
+  po.Register("wav-path", &wav_path, "wav to recognize");
+  po.Register("wav-scp", &wav_scp, "wav.scp path");
+
+  po.Read(argc, argv);
+
+  if (server_port <= 0 || server_port > 65535) {
+    SHERPA_LOG(FATAL) << "Invalid server port: " << server_port;
+  }
+
+  if (po.NumArgs() != 0 || (wav_scp == "" && wav_path == "")) {
+    po.PrintUsage();
+    exit(EXIT_FAILURE);
+  }
+
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  const int32_t sample_rate = 16000;
+  const float interval = 0.02;
+  const int32_t sample_interval = interval * sample_rate;
+
+  std::vector<std::pair <std::string, std::string>> wav_dict;
```

Comment:
Please follow
https://github.com/k2-fsa/sherpa/blob/4087861a8ec41908dae6049eedfae6d75c0fdb0f/sherpa/cpp_api/bin/offline-recognizer.cc#L128
to read `wav.scp`.

Reply:
i use --wav-path or --wav-scp to inform using a wav or wav list.
i think this is a easy way. 
if i use kaldi style, most code will be rewrite

Reply:
There are no conflicts with kaldi's style. You don't need to reinvent the wheel to parse the scp file. Please use the existing code.

Also, please remove `csrc/wav.h`. You don't need that.